app/src/main/java/com/dinhtai/musicapp/view/activity/MainActivity.kt

```diff
@@ -0,0 +1,225 @@
+package com.dinhtai.musicapp.view.activity
+
+import android.Manifest
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.ServiceConnection
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Handler
+import android.os.IBinder
+import android.util.Log
+import android.view.View
+import android.widget.SeekBar
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import com.dinhtai.musicapp.R
+import com.dinhtai.musicapp.model.entity.Music
+import com.dinhtai.musicapp.presenter.MainPresenter
+import com.dinhtai.musicapp.service.ServiceMusic
+import com.dinhtai.musicapp.view.adapter.AdapterMusic
+import com.dinhtai.musicapp.view.adapter.MainViewListener
+import com.dinhtai.musicapp.view.adapter.OnClickItemListener
+import com.dinhtai.musicapp.view.dialog.CustomDialogListMusic
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.item.*
+import java.text.SimpleDateFormat
+
+
+class MainActivity : AppCompatActivity(), OnClickItemListener, View.OnClickListener {
+    private var mainPresenter: MainPresenter? = null
+    private var adapterMusic: AdapterMusic? = null
+    private var recyclerView: RecyclerView? = null
+    private var customDialog: CustomDialogListMusic? = null
+    private var musics: MutableList<Music> = mutableListOf()
+    private lateinit var serviceMusic: ServiceMusic
+    private var isConnect = false
+    private val connectServiceMusic = object : ServiceConnection {
+        override fun onServiceDisconnected(p0: ComponentName?) {
+            isConnect = false
+        }
+
+        override fun onServiceConnected(p0: ComponentName?, p1: IBinder?) {
+            val myBinder = p1 as ServiceMusic.MyBinder
+            serviceMusic = myBinder.getMusicControl()
+            isConnect = true
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        requestPermission()
+        assignViews(img_play, img_show_list, img_next, img_background, img_previous, img_stop)
+        Intent(this, ServiceMusic::class.java).also { intent ->
+            bindService(
+                intent,
+                connectServiceMusic,
+                Context.BIND_AUTO_CREATE
+            )
+        }
```

Comment:
Conventions 2.11 
Em nên tạo hàm getIntent trong service 
còn ở các class khác chỉ gọi ServiceMusic.getIntent(this)